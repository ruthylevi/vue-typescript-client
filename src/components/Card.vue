<template>
    <v-card
    min-width="350"
    max-width="350"
    min-height="200"
    class="mx-auto"
    >
        <v-list-item>
            <v-list-item-avatar tile>
                <v-img :src=avatar></v-img>
            </v-list-item-avatar>
            <v-list-item-content>
                <v-list-item-title class="headline" v-text=title></v-list-item-title>
                <v-list-item-subtitle v-text=advisors.toString()></v-list-item-subtitle>
            </v-list-item-content>
            <v-btn v-if=starred icon>
                <v-icon>mdi-star</v-icon>
            </v-btn >
            <v-btn v-else icon>
                <v-icon>mdi-star-outline</v-icon>
            </v-btn>
        </v-list-item>
        <v-divider></v-divider>
        <v-card-text>
            <div v-if="desc.length < maxChar" v-text="desc"></div>
            <div v-else v-text="desc.substring(0,maxChar)+'...'"></div>
            <v-chip-group column>
                <v-chip label v-for="tag in tags" :key="tag" class="noClick"> {{tag}} </v-chip>
            </v-chip-group>
        </v-card-text>
        <v-card-actions>
            <v-spacer></v-spacer>
            <v-btn text>Read More</v-btn>
        </v-card-actions>
    </v-card>
</template>

<script>
export default {
    props: {
        id: String,
        title: String,
        avatar: String,
        advisors: Array,
        desc: String,
        starred: Boolean,
        tags: Array,
    },
    data: () => ({
        maxChar: 250,
    }),
  }
</script>

<style>
.noClick {
    pointer-events: none;
}
</style>