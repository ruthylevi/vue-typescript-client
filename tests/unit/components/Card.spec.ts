import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { mount } from '@vue/test-utils';
import Card from '@/components/Card.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const id = '00000000-0000-0000-0000-000000000000';
const avatar = 'https://i.imgur.com/bv3cLys.png';
const title = 'marqetplace.';
const starred = true;
const coverImg = '://www.lifeofpix.com/wp-content/uploads/2018/04/275-ake6418-jj-jite-001-1600x1000.jpg';
const tagline = 'The Senior Design Marketplace teamâ€™s mission is to expand on the foundations of the previous groups and implement an online marketplace that will streamline the creation and organization of senior design projects at Stevens. The Senior Design Marketplace is a web application that transforms the senior design project selection and organization process into a simple, efficient, and streamlined activity. Using the marketplace, faculty can create and display projects to students who can choose to view or join them. Students are also able to create their own project proposals and submit them for approval by advisors or program coordinators.';
const tags = ['Computer Science', 'Software Eng.', 'Engineering Management'];

const wrapper = mount(Card, {
  stubs: ['router-link', 'VChipGroup'],
  propsData: {
    id,
    avatar,
    title,
    starred,
    coverImg,
    tagline,
    tags,
  },
});

describe('Card', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Avatar', () => {
    expect(wrapper.find('.v-avatar').exists()).toBe(true);
  });

  test('Title', () => {
    expect(wrapper.find('.headline').text()).toBe(title);
  });

  test('Cover Image', () => {
    expect(wrapper.find('.v-image__image--cover').exists()).toBe(true);
  });

  test('Tagline', () => {
    expect(wrapper.find('.v-card__text>div>div').text()).toBe(tagline);
  });

  test('Tags', () => {
    expect(wrapper.findAll('.v-chip').length).toBe(3);
  });
});
