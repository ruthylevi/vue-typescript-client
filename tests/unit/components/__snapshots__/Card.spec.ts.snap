// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders correctly 1`] = `
<router-link-stub tabindex="0" class="mx-auto v-card v-card--link v-sheet theme--light elevation-2" style="min-height: 350px; min-width: 350px; max-width: 350px;">
  <div class="container pa-1">
    <div class="v-item-group theme--light">
      <div tabindex="-1" class="v-list-item theme--light">
        <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
          <div class="v-responsive v-image">
            <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="v-list-item__content">
          <div class="flex headline">marqetplace.</div>
        </div>
      </div>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <div class="v-responsive v-image" style="height: 194px;">
        <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        <div class="v-responsive__content"></div>
      </div>
      <div class="v-card__text">
        <div>
          <div>The Senior Design Marketplace teamâ€™s mission is to expand on the foundations of the previous groups and implement an online marketplace that will streamline the creation and organization of senior design projects at Stevens. The Senior Design Marketplace is a web application that transforms the senior design project selection and organization process into a simple, efficient, and streamlined activity. Using the marketplace, faculty can create and display projects to students who can choose to view or join them. Students are also able to create their own project proposals and submit them for approval by advisors or program coordinators.</div>
        </div>
        <vchipgroup-stub column=""><span class="noClick v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">Computer Science</span></span><span class="noClick v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">Software Eng.</span></span><span class="noClick v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">Engineering Management</span></span></vchipgroup-stub>
      </div>
    </div>
  </div>
</router-link-stub>
`;
